include/stl2/detail/memory/uninitialized_copy.hpp

```diff
@@ -41,24 +41,39 @@ STL2_OPEN_NAMESPACE {
 		return {std::move(first), std::move(result)};
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// uninitialized_copy [Extension]
+	//
 	template <InputRange Rng, __NoThrowForwardIterator O>
 	requires
-		Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>> &&
-		__ReferenceTo<O, value_type_t<O>>
+		Constructible<value_type_t<O>, reference_t<iterator_t<Rng>>>
 	tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
 	uninitialized_copy(Rng&& rng, O result)
 	{
 		return __stl2::uninitialized_copy(
 			__stl2::begin(rng), __stl2::end(rng), result);
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// uninitialized_copy [Extension]
+	//
+	template <InputRange IRng, class T>
+	requires
+		Constructible<T, reference_t<iterator_t<IRng>>>
+	tagged_pair<tag::in(safe_iterator_t<IRng>), tag::out(iterator_t<ext::span<T>>)>
+	uninitialized_copy(IRng&& irng, ext::span<T> orng)
```

Comment:
Just `span`? Seems a bit undergeneric.

Reply:
...I meant to use `span` as an example for plausible rvalue out ranges, not literally `span` and `span` alone....

Reply:
That explains a lot...